### PR TITLE
Add TypeFloat to field data

### PR DESCRIPTION
**Pull request by vishalnayak**

### CODE REVIEW

File: sdk/framework/field_data_test.go [hunk 1]
```diff
@@ -802,6 +802,39 @@ func TestFieldDataGet(t *testing.T) {
 			"foo",
 			http.Header{},
 		},
+
+		"float type, positive with decimals, as string": {
```
Comment (calvn) [inline sdk/framework/field_data_test.go:806]: Might be also good to include a `"type float, not supplied"` test case (similar to the other cases that tests for fallback to their zero-value).

Comment (vishalnayak) [reply sdk/framework/field_data_test.go:806]: done

**Comment by vishalnayak** [comment]
@mgritter The existing tests didn't cover the "invalid value" cases. I've plumbed it now for new cases to add in those cases as well. As for this PR, I've added one case.

@calvn The feedback you left helped fix a bug. Thanks for that!
